File: user-service/src/main/java/socialapp/userservice/repository/SubscriptionRepository.java
package socialapp.userservice.repository;

import org.springframework.stereotype.Repository;
import socialapp.userservice.model.entity.Subscription;
import socialapp.userservice.repository.common.CommonRepository;

import java.util.Optional;
import java.util.Set;

@Repository
public interface SubscriptionRepository extends CommonRepository<Subscription,Long> {

    Optional<Subscription> findBySenderIdAndReceiverId(Long senderId, Long receiverId);

    Set<Subscription> findAllBySenderId(Long senderId);

    Set<Subscription> findAllByReceiverId(Long receiverId);
    @Override
    default Class<?> entityClass(){
        return Subscription.class;
    };

}

File: user-service/src/main/kotlin/socialapp/ktuserservice/repository/SubscriptionRepository.kt
package socialapp.ktuserservice.repository

import org.springframework.stereotype.Repository
import socialapp.ktuserservice.model.entity.Subscription
import socialapp.ktuserservice.repository.common.CommonRepository

@Repository
interface SubscriptionRepository: CommonRepository<Subscription,Long> {

    fun findBySenderIdAndReceiverId(senderId: Long, receiverId: Long): Subscription?

    fun findAllBySenderId(senderId: Long): Set<Subscription>

    fun findAllByReceiverId(receiverId: Long): Set<Subscription>

    override fun entityClass(): Class<*> = Subscription::class.java
}